bugsnag-android-core/src/main/java/com/bugsnag/android/DeliveryDelegate.java

```diff
@@ -54,23 +54,47 @@ void deliver(@NonNull Event event) {
             }
         }
 
-        if (event.getImpl().getOriginalUnhandled()) {
-            // should only send unhandled errors if they don't terminate the process (i.e. ANRs)
-            String severityReasonType = event.getImpl().getSeverityReasonType();
-            boolean promiseRejection = REASON_PROMISE_REJECTION.equals(severityReasonType);
-            boolean anr = event.getImpl().isAnr(event);
-            if (anr || promiseRejection) {
-                cacheEvent(event, true);
-            } else if (immutableConfig.getAttemptDeliveryOnCrash()) {
-                cacheAndSendSynchronously(event);
+        DeliveryStrategy deliveryStrategy = null;
+        if (event.getDeliveryStrategy() == null) {
+            if (event.getImpl().getOriginalUnhandled()) {
+                String severityReasonType = event.getImpl().getSeverityReasonType();
+                boolean promiseRejection = REASON_PROMISE_REJECTION.equals(severityReasonType);
+                boolean anr = event.getImpl().isAnr(event);
+                if (anr || promiseRejection) {
+                    deliveryStrategy = DeliveryStrategy.STORE_AND_FLUSH;
+                } else if (immutableConfig.getAttemptDeliveryOnCrash()) {
+                    deliveryStrategy = DeliveryStrategy.STORE_AND_SEND;
+                } else {
+                    deliveryStrategy = DeliveryStrategy.STORE_ONLY;
+                }
             } else {
-                cacheEvent(event, false);
+                deliveryStrategy = DeliveryStrategy.SEND_IMMEDIATELY;
+            }
+        }
```

Comment:
This should be in the `Event.getDeliveryStrategy` so that it cannot return `null`